docs/javascripts/extra.js

```diff
@@ -0,0 +1,38 @@
+// When the page is loaded
+document.addEventListener("DOMContentLoaded", () => {
+    // Make all external links open on a new tab
+    // - https://www.w3schools.com/css/css_attribute_selectors.asp
+    let external_links = Array.from(document.querySelectorAll(".md-typeset a[href^=http]"));
+    let repo_link = Array.from(document.querySelectorAll(".md-header__source a[href^=http]"));
+    for (const link of external_links.concat(repo_link)) {
+        link.target = "_blank"
+    }
+
+    // Put <code> around cpp reference links regardless of macros
+    let cpp_reference_links = Array.from(document.querySelectorAll('.md-typeset a[href*="cppreference"]'));
+    let open_std_links = Array.from(document.querySelectorAll('.md-typeset a[href*="open-std"]'));
+    let small_reference_link = Array.from(document.querySelectorAll('.md-typeset a[href*="/small/reference/"]'));
+    for (const link of cpp_reference_links.concat(open_std_links).concat(small_reference_link)) {
+        let already_inside_code_tag = link.parentElement.tagName !== undefined && link.parentElement.tagName.toUpperCase() === 'CODE'
+        let already_has_code_tag = link.firstChild.tagName !== undefined && link.firstChild.tagName.toUpperCase() === 'CODE'
+        let is_code_reference = link.innerHTML.startsWith('std::') || link.href.includes('/small/reference/')
+        if (already_inside_code_tag || already_has_code_tag || !is_code_reference) {
+            continue;
+        }
+        link.innerHTML = "<code>" + link.innerHTML + "</code>"
+    }
+
+    // Fix bug in doxybook reference anchors
+    let brokenAnchors = ['#function-', '#variable-', '#using-']
```

Comment:
This is a silly bug in doxygen2. We shouldn't be fixing it with javascript but this is the best solution we have so far. At some point in the future, we might consider forking doxygen2 and some other alternative.